File: CodeFreak1/ClientApp/src/app/Security/Dtos/roles-permissions-view-model.ts
import { RequestStatus } from "./request-status";
import { RolesViewModel } from "./roles-view-model";
import { PermissionsViewModel } from "./permissions-view-model";

export class RolesPermissionsViewModel extends RequestStatus {
  public Role: RolesViewModel;
  public Permissions: Array<PermissionsViewModel>;
}

File: CodeFreak1/ViewModel/RolesPermissionsViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace CodeFreak1.ViewModel
{
    public class RolesPermissionsViewModel : RequestStatus
    {
        public RolesViewModel Role { get; set; }
        public List<PermissionsViewModel> Permissions { get; set; }

    }
}
